src/Lachain.Core/Config/ConfigManager.cs

```diff
@@ -14,15 +15,40 @@ public class ConfigManager : IConfigManager
         private const ulong _CurrentVersion = 15;
         private IDictionary<string, object> _config;
         public string ConfigPath { get; }
+        public string ConfigBackupPath { get; }
         public RunOptions CommandLineOptions { get; }
 
+        private static readonly ILogger<ConfigManager> Logger = LoggerFactory.GetLoggerForClass<ConfigManager>();
+
+
         public ConfigManager(string filePath, RunOptions options)
         {
             CommandLineOptions = options;
-            var configLoader = new LocalConfigLoader(filePath);
-            _config = new Dictionary<string, object>(configLoader.LoadConfig());
             ConfigPath = filePath;
+            ConfigBackupPath = filePath + ".bak";
+            
+            try {
+                var configLoader = new LocalConfigLoader(ConfigPath);
+                _config = new Dictionary<string, object>(configLoader.LoadConfig());
+                Logger.LogInformation($"Loaded config from {ConfigPath}");
+            }
+            catch (Exception e) {
+                try {
+                    Logger.LogWarning($"Could not load config from {ConfigPath}. Trying from backup ({ConfigBackupPath})");
+                    var configLoader = new LocalConfigLoader(ConfigBackupPath);
```

Comment:
In this case you need to copy backup file to the config file

Reply:
This will be done in the following call to _SaveCurrentConfig()

Reply:
The config will be saved to the config file in the following call to `_SaveCurrentConfig`

Reply:
It is not guaranteed really (for example,  exceptions on the intermediate steps). It is better to copy this file explicitly

Reply:
done in the latest commit.